File: aoc-2024/d01.py
from aocd.models import Puzzle
puzzle = Puzzle(year=2024, day=1).input_data
data = puzzle.split("\n")
a = []
b = []
for d in data:
    x, y = d.strip().split("   ")
    a.append(int(x))
    b.append(int(y))
a.sort()
b.sort()

diff = 0
for x, y in zip(a, b):
    diff += abs(x - y)
print(diff)

common = set(a) & set(b)
sim = 0
for c in common:
    sim += c * b.count(c)
print(sim)

File: aoc-2024/d01.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <sstream>
#include <vector>
#include <cmath>
#include <algorithm>

using namespace std;

int main() 
{
    ifstream file("input.txt");
    string line;
    vector<int> left;
    vector<int> right;
    
    while(getline(file, line))
    {
        istringstream ss(line);
        int x, y;
        ss >> x >> y;
        left.emplace_back(x);
        right.emplace_back(y);
    }
    sort(left.begin(), left.end());
    sort(right.begin(), right.end());
    int diff = 0;
    int sim = 0;
    for (int i = 0; i < left.size(); i++)
    {
        diff += abs(left[i] - right[i]);
        sim += left[i] * count(right.begin(), right.end(), left[i]);
    }
    cout << diff << endl;
    cout << sim << endl;

    file.close();
}
